cmd/soroban-rpc/internal/db/migration.go

```diff
@@ -179,46 +160,48 @@ func (g *guardedMigration) Commit(ctx context.Context) error {
 	if g.alreadyMigrated {
 		return nil
 	}
-	err := setMetaBool(ctx, g.db, g.guardMetaKey, true)
-	if err != nil {
-		return errors.Join(err, g.Rollback(ctx))
-	}
-	return g.db.Commit()
+	return setMetaBool(ctx, g.db, g.guardMetaKey, true)
 }
 
-func (g *guardedMigration) Rollback(_ context.Context) error {
-	return g.db.Rollback()
+func GetMigrationLedgerRange(ctx context.Context, db *DB, retentionWindow uint32) (*LedgerSeqRange, error) {
+	firstLedgerToMigrate := firstLedger
+	latestLedger, err := NewLedgerEntryReader(db).GetLatestLedgerSequence(ctx)
+	if err != nil && !errors.Is(err, ErrEmptyDB) {
+		return nil, fmt.Errorf("failed to get latest ledger sequence: %w", err)
+	}
+	if latestLedger > retentionWindow {
+		firstLedgerToMigrate = latestLedger - retentionWindow
+	}
+	return &LedgerSeqRange{
+		FirstLedgerSeq: firstLedgerToMigrate,
+		LastLedgerSeq:  latestLedger,
+	}, nil
 }
 
-func BuildMigrations(ctx context.Context, logger *log.Entry, db *DB, cfg *config.Config) (Migration, error) {
-	var migrations []Migration
-
-	migrationName := "TransactionsTable"
-	logger = logger.WithField("migration", migrationName)
-	factory := newTransactionTableMigration(
-		ctx,
-		logger,
-		cfg.HistoryRetentionWindow,
-		cfg.NetworkPassphrase,
-	)
+func BuildMigrations(ctx context.Context, logger *log.Entry, db *DB, networkPassphrase string,
+	ledgerSeqRange *LedgerSeqRange,
+) (MultiMigration, error) {
+	migrations := make(MultiMigration, 0, numMigrations)
 
-	m1, err := newGuardedDataMigration(ctx, migrationName, logger, factory, db)
-	if err != nil {
-		return nil, fmt.Errorf("creating guarded transaction migration: %w", err)
+	migrationNameToFunc := map[string]migrationApplierF{
+		transactionsMigrationName: newTransactionTableMigration,
+		eventsMigrationName:       newEventTableMigration,
 	}
```

Comment:
I recommend moving the `migrations` declaration to below `migrationNameToFunc` . That way you can get rid of the `numMigrations` constant:

```
migrations := make(MultiMigration, 0, len(migrationNameToFunc))
```